react-app/testing/setup.ts

```diff
@@ -28,9 +43,23 @@ window.HTMLElement.prototype.scrollIntoView = vi.fn();
 window.HTMLElement.prototype.releasePointerCapture = vi.fn();
 window.HTMLElement.prototype.hasPointerCapture = vi.fn();
 
+vi.spyOn(Auth, "currentAuthenticatedUser").mockImplementation(mockCurrentAuthenticatedUser);
+vi.spyOn(Auth, "userAttributes").mockImplementation(mockUserAttributes);
+vi.spyOn(Auth, "signOut").mockImplementation(async () => {
+  setMockUsername(null);
+});
```

Comment:
This is clever and then you can switch out the user with the env vars right?

Reply:
yes, I borrowed the env idea from the original tests, so `setMockUsername` just sets `process.env.MOCK_USER_USERNAME` to that value and then `mockCurrentAuthenticatedUser` and `mockUserAttributes` use that env